src/test/java/org/json/junit/JSONObjectTest.java

```diff
@@ -4011,5 +4011,23 @@ public static HashMap<String, Object> buildNestedMap(int maxDepth) {
         nestedMap.put("t", buildNestedMap(maxDepth - 1));
         return nestedMap;
     }
+    
 
+    /**
+     * Tests the behavior of the {@link JSONObject} when parsing a bean with null fields
+     * using a custom {@link JSONParserConfiguration} that enables the use of native nulls.
+     * 
+     * <p>This test ensures that uninitialized fields in the bean are serialized correctly
+     * into the resulting JSON object, and their keys are present in the JSON string output.</p>
+     */
+    @Test
+    public void jsonObjectParseNullFieldsWithParserConfiguration() {
+        JSONParserConfiguration jsonParserConfiguration = new JSONParserConfiguration();
```

Comment:
Please add a default case where the JSONObject is invoked without a JSONParserConfiguration (or with NativeNulls disabled), and assert that the resulting object is empty. This will help clarify how the flag changes the behavior.